crates/bevy_ecs/src/query/iter.rs

```diff
@@ -346,15 +346,10 @@ where
     }
 }
 
-// NOTE: We can cheaply implement this for unfiltered Queries because we have:
-// (1) pre-computed archetype matches
-// (2) each archetype pre-computes length
-// (3) there are no per-entity filters
-// TODO: add an ArchetypeOnlyFilter that enables us to implement this for filters like With<T>.
-// This would need to be added to all types that implement Filter with Filter::IS_ARCHETYPAL = true
-impl<'w, 's, Q: WorldQuery, QF> ExactSizeIterator for QueryIter<'w, 's, Q, QF, ()>
+impl<'w, 's, Q: WorldQuery, QF, F> ExactSizeIterator for QueryIter<'w, 's, Q, QF, F>
```

Comment:
Would it also be possible to implement `ExactSizeIterator` on `QueryCombinationIter`? I think calling `self.size_hint().0` (or `self.size_hint().1.unwrap()` for  a more fool-proof version) would work.

Reply:
I don't know if it is possible actually, since originally `ExactSizeIterator` is only implemented for `QueryIter`.  I will try to implement this when I am done with the tests.